sdk/src/main/java/io/dapr/actors/DaprHttpAsyncClient.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+package io.dapr.actors;
+
+import okhttp3.*;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.UUID;
+
+/**
+ * Http client to call Dapr's API for actors.
+ */
+class DaprHttpAsyncClient implements DaprAsyncClient {
+
+    /**
+     * Defines the standard application/json type for HTTP calls in Dapr.
+     */
+    private static final MediaType MEDIA_TYPE_APPLICATION_JSON = MediaType.get("application/json; charset=utf-8");
+
+    /**
+     * Shared object representing an empty request body in JSON.
+     */
+    private static final RequestBody REQUEST_BODY_EMPTY_JSON = RequestBody.create(MEDIA_TYPE_APPLICATION_JSON, "");
+
+    /**
+     * Base Url for calling Dapr. (e.g. http://localhost:3500/)
+     */
+    private final String baseUrl;
+
+    /**
+     * Http client used for all API calls.
+     */
+    private final OkHttpClient httpClient;
+
+    /**
+     * Creates a new instance of {@link DaprHttpAsyncClient}.
+     * @param baseUrl Base Url for calling Dapr. (e.g. http://localhost:3500/)
+     * @param httpClient RestClient used for all API calls in this new instance.
+     */
+    DaprHttpAsyncClient(String baseUrl, OkHttpClient httpClient)
+    {
+        this.baseUrl = baseUrl;
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<String> invokeActorMethod(String actorType, String actorId, String methodName, String jsonPayload) {
+        String url = String.format(Constants.ACTOR_METHOD_RELATIVE_URL_FORMAT, actorType, actorId, methodName);
+        return invokeAPI("PUT", url, jsonPayload);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<String> getState(String actorType, String actorId, String keyName) {
+        String url = String.format(Constants.ACTOR_STATE_KEY_RELATIVE_URL_FORMAT, actorType, actorId, keyName);
+        return invokeAPI("GET", url, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> saveState(String actorType, String actorId, String keyName, String data) {
+        String url = String.format(Constants.ACTOR_STATE_KEY_RELATIVE_URL_FORMAT, actorType, actorId, keyName);
+        return invokeAPIVoid("PUT", url, data);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> removeState(String actorType, String actorId, String keyName) {
+        String url = String.format(Constants.ACTOR_STATE_KEY_RELATIVE_URL_FORMAT, actorType, actorId, keyName);
+        return invokeAPIVoid("DELETE", url, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> saveStateTransactionally(String actorType, String actorId, String data) {

```

Comment:
We can remove this saveState, it will not be used as Actors will always save transactionally,

Reply:
Done.